src/java/org/apache/cassandra/db/compaction/ShardTracker.java

```diff
@@ -54,6 +57,10 @@ public interface ShardTracker
 
     double rangeSpanned(PartitionPosition first, PartitionPosition last);
 
+    /**
+     * The index of the shard this tracker is currently on. By convention, the shard index starts at -1 for the
+     * minimum token and increases by one for each shard.
```

Comment:
I just noticed this javadoc isn't right and we have two ways that we count index values. In `ShardManagerTrivial`'s anonymous class, we return 0 always. In `BoundaryTracker`, we start at 0 and increment. In `BoundaryTrackerDiskAware`, we return -1 immediately after creation, but any skips will trigger it to go to 0.

My main point of confusion is what the shard index is. It's not the index in the backing array of split points, it's the index of the shard count. At the moment, it's only used for logging, so the stakes are fairly low here. I will try to find a conclusive answer tomorrow.